test/dotnet.Tests/CliSchemaTests.cs

```diff
@@ -0,0 +1,1133 @@
+        "minimum": 1,
+        "maximum": 1
+      },
+      "required": false,
+      "recursive": false
+    },
+    "--self-contained": {
+      "description": "Publish the .NET runtime with your application so the runtime doesn't need to be installed on the target machine.\r\nThe default is 'false.' However, when targeting .NET 7 or lower, the default is 'true' if a runtime identifier is specified.",
+      "hidden": false,
+      "aliases": [
+        "--sc"
+      ],
+      "helpName": null,
+      "valueType": "System.Boolean",
+      "hasDefaultValue": false,
+      "defaultValue": null,
+      "arity": {
+        "minimum": 0,
+        "maximum": 1
+      },
+      "required": false,
+      "recursive": false
+    },
+    "--source": {
+      "description": "",
+      "hidden": true,
+      "aliases": [],
+      "helpName": "SOURCE",
+      "valueType": "System.Collections.Generic.IEnumerable<System.String>",
+      "hasDefaultValue": false,
+      "defaultValue": null,
+      "arity": {
+        "minimum": 1,
+        "maximum": null
+      },
+      "required": false,
+      "recursive": false
+    },
+    "--use-current-runtime": {
+      "description": "Use current runtime as the target runtime.",
+      "hidden": false,
+      "aliases": [
+        "--ucr"
+      ],
+      "helpName": null,
+      "valueType": "System.Boolean",
+      "hasDefaultValue": false,
+      "defaultValue": null,
+      "arity": {
+        "minimum": 0,
+        "maximum": 0
+      },
+      "required": false,
+      "recursive": false
+    },
+    "--verbosity": {
+      "description": "Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].",
+      "hidden": false,
+      "aliases": [
+        "-v"
+      ],
+      "helpName": "LEVEL",
+      "valueType": "Microsoft.DotNet.Cli.VerbosityOptions",
+      "hasDefaultValue": false,
+      "defaultValue": null,
+      "arity": {
+        "minimum": 1,
+        "maximum": 1
+      },
+      "required": false,
+      "recursive": false
+    },
+    "--version-suffix": {
+      "description": "Set the value of the $(VersionSuffix) property to use when building the project.",
+      "hidden": false,
+      "aliases": [],
+      "helpName": "VERSION_SUFFIX",
+      "valueType": "System.String",
+      "hasDefaultValue": false,
+      "defaultValue": null,
+      "arity": {
+        "minimum": 1,
+        "maximum": 1
+      },
+      "required": false,
+      "recursive": false
+    }
+  },
+  "subcommands": {}
+}
+""";
+
+    public static TheoryData<string[], string> CommandsJson => new()
+    {
+        { new[] { "solution", "list", "--cli-schema" }, SolutionListJson },
+        { new[] { "clean", "--cli-schema" }, CleanJson },
+        { new[] { "reference", "--cli-schema" }, ReferenceJson },
+        { new[] { "workload", "install", "--cli-schema" }, WorkloadInstallJson },
+        { new[] { "build", "--cli-schema" }, BuildJson }
+    };
```

Comment:
The test is very simple and just checks against the output of these 5 commands.
- `dotnet solution list`
- `dotnet clean`
- `dotnet reference`
- `dotnet workload install`
- `dotnet build`

Since the SDK version is in the output, I had to inject the built version into the hard-coded strings so that the test will still work as the version changes.